### ISSUE TITLE
Get rid of `variable` in kernels, and properly name this parameter `equation`

### ISSUE THREAD

**Issue by GiudGiud** [idaholab/moose#24524]
## Reason
It's a confusion that has lived on for 15 years. We have to explain it over and over again
It's a barrier to understanding MOOSE in depth, a gotcha that does not need to be

## Design
Massive parameter rename, deprecation of 100% of all moose inputs
We could use the name of the variable to name the equation at first. 

## Impact
Improve user understanding of MOOSE by a few per cent.
Lessen confusion for multiphysics users

**Comment by lindsayad** [comment]
@idaholab/moose-ccb

**Comment by cticenhour** [comment]
Considering that kernels reflect the outcome of an operator performing its operation on a particular variable, with potentially several variables being referenced in a particular equation (albeit one being the "primary" variable in a given PDE), I'm not sure I like the suggested swap in the title here. 

You want an `equation` to be the object-level reference to its primary variable, with coupled variable parameters being expressed as variable parameters the same as they are now? I think this adds confusion and complexity.

**Comment by joshuahansel** [comment]
In those cases where you're having to duplicate a class to use something other than the equation variable in a term, you can just add another parameter `variable_to_use_in_some_term` and achieve the same effect as what you suggest, right? Then you should be able to delete the redundant class? Then it's just a matter of whether the parameter names are intuitive.

I agree that "variable" may not always be intuitive (maybe "equation_variable", "residual_variable", or "associated_variable" might be more descriptive names), but I think this change is a bit extreme compared to the potential clarification.

Perhaps you can give some specific examples?

**Comment by joshuahansel** [comment]
> but then effectively, you are using a variable parameter which really means equation.

Yeah, I agree this would be better named as something more descriptive, but I think "variable" is not terrible at least...

For `CoupledForce`, I recommend renaming `v` to `force_variable`, or maybe just `force` (a Functor value) 😈 There's also the possibility of using the parameter renaming thing to change "variable" to "equation_variable" in one class if not clear, right? I don't know that it would be a good practice to do this (probably not), but it's a consideration.

**Comment by lindsayad** [comment]
I think that having `variable` somewhere in the name does have value as you will be supplying a parameter value corresponding to a variable. Something like `equation_variable` like @joshuahansel suggested

**Comment by GiudGiud** [comment]
It doesn't have to be a `variable`'s name if we introduce the concept/system of `Equation`.
I could see:
```
[Variables/T]
[]
[AuxVariables/q]
[]
[Equations]
  [heat_conduction]
    primary_variable = T
    coupled_variables = q
  []
[]
[Kernels (or whatever the new name ^^)]
  [diff]
    type = Diffusion
    equation = heat_conduction
    variable = T
  []
  [source]
    type = HeatSource
    equation = heat_conduction
    value = q
  []
[]
```
very nicely, we stop referring to T in the heat source so there's no way to get it wrong. Because the HeatSource kernel is not actually operating on variable T, in the context of a coupled multiphysics problem it s operating on more than that one variable (here it s just T, and even here, it acts on residuals/jacobian not T)

**Comment by lindsayad** [comment]
In order for users to build effective multiphysics calculations, they often need to think about the disparate scaling between physics residuals and jacobians. Whatever we do should not decrease the amount they think about this. I'm not saying what you're proposing does that.

In the example of a residual that is like X * Y, sure the decision about what variable to add the residual and Jacobian to is highly arbitrary, but the decision has real significant meaning depending on the residual/Jacobian scale compared to whatever terms exist for the chosen variable/equation
